### PR TITLE
MAINT: 1.9.2 backports

### PR DESCRIPTION
**Pull request by tylerjereddy**
Try to get ready for SciPy `1.9.2` and release of Python `3.11` (and other) wheels via `cibuildwheel` infrastructure. Merge conflicts related to build system infrastructure were extreme--any PR with a backport label that modified code that doesn't even exist on the maintenance branch had its backport label stripped and a comment added to the PR explaining why.

In cases where I saw that I could cleanly apply a patch to the build system, but the context in flanking code had changed radically, I applied the patch nonetheless--for example, many meson data structures were using things like `py3_dep` on maintenance branch but neither before nor after on the labeled backport PRs--in these cases I apply the patch directly and assume that the context change is safe (otherwise the burden of reinterpretation of the code would be huge).

Also, remember that this branch still uses `do.py` not `dev.py`, to give you an idea of how much stuff has drifted..

Backports currently included (those excluded should have a recent comment from me, I'll go back and check notifications to see what else needs to be backported... there will be more..):

- gh-16628
- gh-16842
- gh-16909
- gh-16934
- gh-16943
- gh-16954
- gh-16969
- gh-16975
- gh-16991
- gh-17011
- gh-17013
- gh-17046
- gh-17064
- gh-17096
- gh-17127
- gh-17131

### CODE REVIEW

File: doc/release/1.9.2-notes.rst [hunk 1]
```diff
@@ -5,15 +5,60 @@ SciPy 1.9.2 Release Notes
 .. contents::
 
 SciPy 1.9.2 is a bug-fix release with no new features
-compared to 1.9.1.
+compared to 1.9.1. It also provides wheel for Python 3.11
+on several platforms.
```
Comment (tylerjereddy) [inline doc/release/1.9.2-notes.rst:9]: last I checked we still have a certificate issue for MacOS python 3.11 ARM and x86, is there something I can backport for that?

File: scipy/optimize/meson.build [hunk 2]
```diff
@@ -140,7 +134,8 @@ cobyla_module = custom_target('cobyla_module',
 
 _cobyla = py3.extension_module('_cobyla',
   [cobyla_module, fortranobject_c, 'cobyla/cobyla2.f', 'cobyla/trstlp.f'],
-  c_args: [numpy_nodepr_api, '-Wno-unused-variable'],
+  c_args: [numpy_nodepr_api, Wno_unused_variable],
+  [cobyla_module, 'cobyla/cobyla2.f', 'cobyla/trstlp.f'],
```
[the remark this answers is not in the document]
Comment (tylerjereddy) [reply scipy/optimize/meson.build:138]: or it should replace the one above, I'll have to double check if it was genuinely in a diff or change from non-backport PR context

File: scipy/integrate/meson.build [hunk 3]
```diff
@@ -160,6 +150,7 @@ py3.extension_module('_lsoda',
   link_with: [lsoda_lib, mach_lib],
   c_args: [numpy_nodepr_api, '-Wno-unused-variable'],
   dependencies: [py3_dep, lapack],
+  c_args: [numpy_nodepr_api, Wno_unused_variable],
```
Comment (tylerjereddy) [inline scipy/integrate/meson.build:153]: this should replace the `c_args` above, the string style was superseded in backports IIRC

**Comment by mdhaber** [comment]
@tylerjereddy gh-16814 just merged, can we add it here? Shall I open a PR against your branch, push here, or let you do it?

**Comment by tylerjereddy** [comment]
Eh, there are still `meson` build issues, but I've cleaned things up a little at least. I'm still not fully convinced on the safety of the backports vs. the degree of change around those backports, but I'll try to take another look tomorrow. Some `pyproject.toml` stuff may need to be reverted as well, wheel issue is familiar...

File: doc/release/1.9.2-notes.rst (hunk 1 above)
Comment (rgommers) [reply doc/release/1.9.2-notes.rst:9]: It should disappear, the fix was merged 3 days ago: https://github.com/actions/python-versions/pull/189

**Comment by rgommers** [comment]
```
FAILED ../../build-install/lib/python3.10/site-packages/scipy/linalg/tests/test_lapack.py::test_gges_tgexc[float32]
FAILED ../../build-install/lib/python3.10/site-packages/scipy/linalg/tests/test_decomp.py::TestQZ::test_qz_single
```
That requires backporting the relevant fix from gh-17057.

The doc build is failing on a single Matplotlib deprecation warning for `set_tight_layout`. Not sure why it's not erroring on `main`, but that one can be solved by pinning Matplotlib or updating the code with the suggested replacement.

I will have a look at the build failures and push fixes for them.

**Comment by rgommers** [comment]
> Eh, there are still `meson` build issues, but I've cleaned things up a little at least. I'm still not fully convinced on the safety of the backports vs. the degree of change around those backports, but I'll try to take another look tomorrow.

I think it's pretty close now. It is quite a bit of churn, but I don't see a good alternative for Python 3.11 unless we pull the 1.10.0 release way forward - which is likely a lot more work.

**Comment by rgommers** [comment]
Okay, all the regular CI should work now, and I also pushed a change to CI triggering for `wheels.yml`, so the wheel builders are running now on this PR when you add `[wheel build]` to the commit message.

If this is all green, I think the PR is ready to merge, and after that it needs one more commit to set the version number to `1.9.2` and push the tag.

**Comment by tylerjereddy** [comment]
Interesting, did you fix the MacOS Python 3.11 certificate issue somehow or did that just go away with a dependency update somewhere?

**Comment by rgommers** [comment]
> Interesting, did you fix the MacOS Python 3.11 certificate issue somehow or did that just go away with a dependency update somewhere?

See https://github.com/scipy/scipy/pull/17132#discussion_r986682047

**Review by rgommers — APPROVED** [review]
Even aarch64 on QEMU seems happy. The one thing that this PR does not yet do is uncomment the `Upload wheels` section of `wheels.yml`. I think that can be done separately - or in this PR without re-testing. Let's just confirm that the code is identical there between `main` and `1.9.x`.

**Comment by mdhaber** [comment]
Can we get these two bug fixes in? https://github.com/scipy/scipy/pull/17132#issuecomment-1264786435

**Comment by tylerjereddy** [comment]
@mdhaber I may do those separately but will try to get them in--the diff here is getting close to 4,000 lines. Let me try to skim this PR over one more time then I'll probably merge it based on Ralf's recommendation/fixes.

**Comment by mdhaber** [comment]
Great, thanks!

**Pull request merged**